File: 2022.1/Programação Orientada a Objetos/lista1python/Ex8.py
def f(x):
    a = 1
    b = -1
    c = -13
    d = 8

    return (a*x*x*x + b*x*x + c*x +d)

a = float(input())
b = float(input())

c = (a+b)/2
contador = 0

while(abs(a-b) > 0.0000001):

    if(f(c) * f(a) < 0):
        b = c
        c = (a+b)/2
    else:
        a = c
        c = (a+b)/2

    contador += 1

print("A raiz encontrada no intervalo dado foi:", c)
print("Foram realizadas", contador, "iterações")

File: 2022.1/Programação Orientada a Objetos/lista1java/Ex8.java
public class Ex8{
    public static double f(double x){
        int a = 1;
        int b = -1;
        int c = -13;
        int d = 8;

        return (a*x*x*x + b*x*x + c*x + d);
    }
    
    public static void main(String args[]) throws Exception{

        double a = EntradaTeclado.leDouble();
        double b = EntradaTeclado.leDouble();
        
        double c = (a + b)/2;
        int contador = 0;

        while(Math.abs(a-b) > 0.0000001){
    
            if(f(c) * f(a) < 0){
                b = c;
                c = (a + b)/2;
            }
            else{
                a = c;
                c = (a + b)/2;
            }

            contador++;
        }

        System.out.println("A raiz encontrada no intervalo dado foi: " + c);
        System.out.println("Foram realizadas " + contador + " iterações");
    }
}
